### ISSUE TITLE
Winget Inspiration, parallel Installation and output 

### ISSUE THREAD

**Issue by Karl-WE** [microsoft/winget-cli#2824]
### Description of the new feature / enhancement

Lately read about this apt alternative called nala, and thought it would be great to have a look on some features including but not limited to parallel Installation (limited in Windows with (repackaged) msi, but not with other Installation method such as inno or Appx / msix, I believe.

Also the output is notable. 
Thinking about winget's role in terms of Intune app deployment nala also provides a good example on visualizing dependencies. 

https://www-omgubuntu-co-uk.cdn.ampproject.org/v/s/www.omgubuntu.co.uk/2023/01/install-nala-on-ubuntu/

### Proposed technical implementation details

Unfortunately cannot provide a concise description other than kindly asking herewith @denelon and the team to have a look and be inspired.

**Comment by Karl-WE** [comment]
Parallel Installation #225

**Comment by Karl-WE** [comment]
Package Dependencies #163

**Comment by denelon** [comment]
I like the view. I think we can adopt something like this. Thanks for linking the other issues!

**Comment by Masamune3210** [comment]
Most installers do not allow concurrent installs, especially any based on MSI. What can be worked on is being worked on, but we cant really help it if the underlying installer tech doesn't support it

**Comment by Karl-WE** [comment]
I believe on in this PR we should focus on the matter of Winget output to the console

I can share my own experience with nala and it is truely amazing what they accomplished. It might look fancy due to the colors, we don't have to adapt these, but it is very informative.

In addition tried out snap and the snap store which is also very inspiring in terms of how Microsoft Store / Intune winget distribution and winget could work together.

**Comment by JABirchall** [comment]
> Most installers do not allow concurrent installs, especially any based on MSI. What can be worked on is being worked on, but we cant really help it if the underlying installer tech doesn't support it

The download can atleast be parrallized as a start. I understand some installers may have conflicts. But most dont.

**Comment by denelon** [comment]
We've been looking both at the "download packages in parallel" for install/upgrade scenarios with more than one package, as well as the potential for multiple installs. 

The MSIX installer will allow up to six concurrent installs and shouldn't interfere with any other installs. 

MSI is limited to one (which can also be paused for Windows updates). 

The world of .exe-based installers is where the unknown risk lies. We would likely limit those to one at a time to avoid problems they could cause. 

I don't believe it makes sense to allow different installer types to run concurrently.

When we get into these types of scenarios, we will need to build a graph for what to install based on the installer type and the dependencies installer types. Then we can look at some performance optimizations.

**Comment by Karl-WE** [comment]
Thank you @denelon.

This was quite expected. From my experience in the field exe based installers could run in parallel mode.
But that's limited to inno based one like VLC. 

Installshield and other often are just wrappers and often enough use msi.

Even msstore some are just wrappers, example Spotify that will download a payload installer like msi. 

There could be a way to allow it based on installer type specified in the yaml but that's extremely vague and prone to fail due to msi will not wait on each other but often fail.

As my PR wasn't clean having just one idea at a time should we split it into another one for the visual output idea?

**Comment by denelon** [comment]
Yes, it's easier to track and make enough progress to close on single requests. I still try to honor the author's intent, but it's harder to track when all the work is done. 😊

**Comment by Karl-WE** [comment]
@denelon I have missed your comment, sorry for necro. This PR is now just about parallel installation.

**Comment by nkomarov** [comment]
> The download can atleast be parrallized as a start. I understand some installers may have conflicts. But most dont.

It's an easy subtask to work on, think there should be an issue about it. Even if installs can only happen one after another, we could at lest avoid unnecessary pauses because the next download only starts as soon as the previous installation has been completed.
